contrib/local-setup

```diff
@@ -1,22 +1,61 @@
 #!/bin/bash
 
-set -e
-
-DIR="$(git rev-parse --show-toplevel)"
-cd "$DIR"
-
-# check if clang-format is installed
-if [ -z "`which clang-format`" -o -z "`which git-clang-format`" ]; then
-	echo "Please install clang-format."
-	echo "If using Mac OS X, use 'brew install clang-format'."
-	echo "Setup not successful."
-	exit 1
-fi
-
-# add configuration for git-clang-format
-git config --local --add clangFormat.style file
-git config --local --add clangFormat.extension hpp,cpp
-
-# install pre-commit hook
-cp -p contrib/pre-commit.hook .git/hooks/pre-commit
-chmod +x .git/hooks/pre-commit
+(
+	# create a subshell to avoid bothering the user with changing the
+	# current directory.
```

Comment:
I think this might be a misunderstanding. This is a bash program. You should run it by calling it, not sourcing it. Sourcing a script will generally change your environment, while called programs are always executed in a subshell.

Reply:
Yes, this makes sense. Right now, I can not understand why I put that there when I wrote that code.